src/tree/v2/ResourceTreeDataProviderBase.ts

```diff
@@ -0,0 +1,97 @@
+/*---------------------------------------------------------------------------------------------
```

Comment:
How do we feel about moving the files in `src/tree/v2/*` to `src/tree/v2/common`? I think it's a bit cleaner and more organized.

<img width="349" alt="image" src="https://user-images.githubusercontent.com/12476526/190530430-cd61f1bd-97ec-47ce-a70a-afa2fc289efd.png">

<img width="339" alt="image" src="https://user-images.githubusercontent.com/12476526/190531054-f21babca-723a-4871-bde1-c8ecd88b4773.png">

Reply:
I've always preferred to keep more common types in the parent folder, with more specific types in child folders.  I suppose that's more C# style (due to its automatic namespacing), which would otherwise be awkward if the common types ended up in a sibling folder.  I feel like the suggestion might result in a number of `common` folders, which could be confusing, or require a bunch of thought about whether any given type belongs in the parent or a common folder.